### ISSUE TITLE
🐛 [Bug] Quest Gameobject click make client kicked 

### ISSUE THREAD

**Issue by betayi** [vmangos/core#1469]
## 🐛 Bug report

Quest  409 Proving Allegiance
Player click the candle box
then client kicked 
even if you add this item in you bag
you still cannot click the quest gamobject “Lillith's Dinner Table” to continue this quest.

server log:
get value from nonexistent value field: 147 (count: 26) for object typeid: 5 type mask: 33 
/root/vmangos/core/src/game/Objects/Object.h:390: Error: Assertion in HasFlag failed: index < m_valuesCount || PrintIndexError(index ,false)

[Anticrash] Player XXXXXXX  Cheat: ASSERT failed, Penalty: Kicked. 
### Expected behavior

client should not be click and player should get quest item 3080
### Steps to reproduce

1 . go creature id 1497
2 . q a 409
3. click gameobject candle box on ground
4. watch the console screen
5. additem 3080
6. click the quest gameobject Lillith's Dinner Table at near
7. watch the console screen 

### Version & Environment

Client Version: 
1.12.1

Commit Hash:  
cc8da9f2cd10f419f1ca

OS Client:  Win10
OS Server:  Ubuntu 20.4

### Crashlog

- None

**Comment by betayi** [comment]
And The same break quest with gameobject click : 477 Border Crossings

**Comment by Oroxzy** [comment]
i can't reproduce this, it works fine for me.

**Comment by betayi** [comment]
@Oroxzy  use the same commit version ? 
it's so weired, 
i will test more gameobject

**Comment by betayi** [comment]
Well， I can confirm all gameobject that used by quest start or end, will cause this exception.

**Comment by ratkosrb** [comment]
I can't reproduce it either. There are no errors when doing the quest.

**Comment by rosen40** [comment]
[Anticrash] Player XXXXXXX Cheat: ASSERT failed, Penalty: Kicked.

I might be wrong, but isn't this the anti cheat ^ ? 

Maybe you are forcibly accepting a quest that has higher required level than your character is with gm command and AC flags it as cheating?

**Comment by ratkosrb** [comment]
No, anticrash is different thing. When an exception occurs inside the packet handler, the server kicks that player for sending badly formatted packets. In this case something directly called by the packet handler tries to access non-existent update field. But as i said again, i cannot reproduce the problem. I think he has custom changes that caused this.

**Comment by ratkosrb** [comment]
I just looked at his fork and this is 100% his fault. I found the custom code he added which causes it. He checks unit npc flags on a WorldObject pointer, which is not guaranteed to be a creature, and thus when turning in quests to GameObjects his core crashes, since GameObjects do not have Unit update fields. The vmangos issues page is not a place to ask for help with your custom changes. Do not open issues that have not been verified on a clean setup. You are responsible for the quality of your own code. I do not provide support or assistance with custom stuff.
